src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+                    Logging.PrintInfo(Logging.Web, this, SR.net_log_finding_matching_certs);
+            }
+
+            //
+            // ATTN: When the client cert was returned by the user callback OR it was guessed AND it has no private key,
+            //       THEN anonymous (no client cert) credential will be used.
+            //
+            // SECURITY: Accessing X509 cert Credential is disabled for semitrust.
+            // We no longer need to demand for unmanaged code permissions.
+            // EnsurePrivateKey should do the right demand for us.
+            for (int i = 0; i < filteredCerts.Count; ++i)
+            {
+                clientCertificate = filteredCerts[i] as X509Certificate;
+                if ((selectedCert = EnsurePrivateKey(clientCertificate)) != null)
+                {
+                    break;
+                }
+
+                clientCertificate = null;
+                selectedCert = null;
+            }
+
+            GlobalLog.Assert(((object)clientCertificate == (object)selectedCert) || clientCertificate.Equals(selectedCert), "AcquireClientCredentials()|'selectedCert' does not match 'clientCertificate'.");
+
+            GlobalLog.Print("SecureChannel#" + Logging.HashString(this) + "::AcquireClientCredentials() Selected Cert = " + (selectedCert == null ? "null" : selectedCert.Subject));
+            try
+            {
+                // Try to locate cached creds first.
+                //
+                // SECURITY: selectedCert ref if not null is a safe object that does not depend on possible **user** inherited X509Certificate type.
+                //
+                byte[] guessedThumbPrint = selectedCert == null ? null : selectedCert.GetCertHash();
+                SafeFreeCredentials cachedCredentialHandle = SslSessionsCache.TryCachedCredential(guessedThumbPrint, _protocolFlags, _encryptionPolicy);
+
+                // We can probably do some optimization here. If the selectedCert is returned by the delegate
+                // we can always go ahead and use the certificate to create our credential
+                // (instead of going anonymous as we do here).
+                if (sessionRestartAttempt && cachedCredentialHandle == null && selectedCert != null)
+                {
+                    GlobalLog.Print("SecureChannel#" + Logging.HashString(this) + "::AcquireClientCredentials() Reset to anonymous session.");
+
+                    // IIS does not renegotiate a restarted session if client cert is needed.
+                    // So we don't want to reuse **anonymous** cached credential for a new SSL connection if the client has passed some certificate.
+                    // The following block happens if client did specify a certificate but no cached creds were found in the cache.
+                    // Since we don't restart a session the server side can still challenge for a client cert.
+                    if ((object)clientCertificate != (object)selectedCert)
```

Comment:
Won't fix. Similar behavior different syntax.
Changing in one place doesn't make sense. Changing everywhere in the code is out of scope.